### PR TITLE
PLAT-765-1 Migrated Some AJAX Javascript Engine Code to Typescript

### PR DESCRIPTION
**Pull request by oliver-richers**
- migrated ActionQueue.js
- migrated KeepAlive.js
- removed unused Fading.js
- removed unused Mutex.js
- removed unused method enableInput in DOMContext.js

Test Plan:
- startet test servlet, opening pages, popups, etc.
- grep for the following texts to ensure proper renaming:
  - 'AQ_', 'KeepAlive', 'Mutex', 'Fade', 'Fading', 'enableInput'
  - find src/ -type f | egrep 'js$|ts$|java$' | xargs grep -in AQ_

### CODE REVIEW

File: platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/ForwardDeclarations.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+
+declare let SESSION_TIMED_OUT: boolean;
+
+declare function SR_sendAjaxRequest(message: any, form: any): void;
```
Comment (alexander-schmidt) [inline platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/ForwardDeclarations.ts:4]: That `SR_` seems cryptic. When we're already at it: Can we either remove the prefix, or spell it out?

File: platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/KeepAlive.ts [hunk 2]
```diff
@@ -1,52 +1,57 @@
...
-	};
+			KEEP_ALIVE.schedule();
+		}
+	}
+}
 
-	var m_delay = delay;
-	var m_handler = null;
-};
+let KEEP_ALIVE = new KeepAlive(3*60*1000);
```
Comment (alexander-schmidt) [inline platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/KeepAlive.ts:57]: We should probably extract a constant here.

File: platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/Actions.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
+
+interface Action {
+	execute(): void;
+}
+
+// This action represents a server request. When executed, this
```
Comment (alexander-schmidt) [inline platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/Actions.ts:6]: In most of the other source files, we mimic...
```
/**
 * Javadoc syntax
 */
```
We should probably do the same in TS-transcribed code.
Unless this was a deliberate change of style.

**Review by alexander-schmidt — CHANGES_REQUESTED** [review]

File: platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/ForwardDeclarations.ts (hunk 1 above)
Comment (oliver-richers) [reply platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/ForwardDeclarations.ts:4]: not in this PR, as you can see, it is only a forward declaration of a "module" I did not want to touch, yet.

File: platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/KeepAlive.ts (hunk 2 above)
Comment (oliver-richers) [reply platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/KeepAlive.ts:57]: I can do that.

File: platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/Actions.ts (hunk 3 above)
Comment (oliver-richers) [reply platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/Actions.ts:6]: I have to check what is the recommended type of comments in TypeScript.

Comment (oliver-richers) [reply platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/Actions.ts]: converted all comments to TypeDoc

File: platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/KeepAlive.ts (hunk 2 above)
Comment (oliver-richers) [reply platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/KeepAlive.ts]: did so

**Review by alexander-schmidt — APPROVED** [review]

**Pull request merged**